R/utilities-quantity.R

```diff
@@ -155,8 +155,11 @@ setQuantityValuesByPath <- function(quantityPaths, values, simulation, units = N
     path <- enc2utf8(quantityPaths[[i]])
     value <- values[[i]]
     if (!is.null(units)) {
-      dimension <- rClr::clrCall(task, "DimensionNameByPath", simulation$ref, path)
-      value <- toBaseUnit(quantityOrDimension = dimension, values = value, unit = units[[i]])
+      dimension <- rClr::clrCall(task, "DimensionNameByPath", simulation$ref, path, stopIfNotFound)
+      # Dimension ca be be empty if the path was not found and stopIfNotFound is false
+      if (dimension != "" || stopIfNotFound) {
```

Comment:
I am still not quite sure I understand this :D

I think you should exit early, if dimension is `""`, than the path is definitely not found and no need to continue:

```
    if (!is.null(units)) {
      dimension <- rClr::clrCall(task, "DimensionNameByPath", simulation$ref, path, stopIfNotFound)
      # Dimension ca be be empty if the path was not found and stopIfNotFound is false
      if (dimension == "") {
        next
      }
      value <- toBaseUnit(quantityOrDimension = dimension, values = value, unit = units[[i]])
    }
```

Reply:
done